processor/spanmetricsprocessor/processor.go

```diff
@@ -234,13 +249,13 @@ func (p *processorImp) ConsumeTraces(ctx context.Context, traces pdata.Traces) e
 		return err
 	}
 
+	p.reset()
```

Comment:
Are you sure resetting here will not affect the metrics in `m`? If there are reference types I would want to be sure that they wouldn't be affected in the following line.

Reply:
Yes, I am sure the `m` should not be impacted. `m` is a pointer but it's not held by the processor.

I have moved it to a defer function so that we can make the downstream components get the result a little bit faster.

Reply:
Actually, kind of concerned with this `reset` since it's not under any lock. There was a reason that I previously reset exemplar data only inside `buildMetrics()`. Would be nice to have this concurrency verified by tests too